presto-main/src/main/java/com/facebook/presto/sql/planner/SqlPlannerContext.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+public class SqlPlannerContext
+{
+    private int cteDepth;
```

Comment:
I wonder if we should make it more general so we can add other properties.

Reply:
I realized that the depth of CTE is not the (only) problem, it's the combination of join of previous tables in CTE and long chain of CTE. Hence I update the diff to track the number of nodes, rather than CTE depth here.

> I wonder if we should make it more general so we can add other properties.

Could you elaborate? I assume we can always add new fields to the context when we found necessary?